add kuromoji_completion token filter docs #8151 (#8476)
opensearch-project/documentation-website: _analyzers/token-filters/kuromoji-completion.md

---
layout: default
title: Kuromoji completion
parent: Token filters
nav_order: 230
---

# Kuromoji completion token filter

> The `kuromoji_completion` token filter is used to stem Katakana words in Japanese, which are often used to represent foreign words or loanwords. This filter is especially powerful for autocompletion or suggest queries, where partial matches on Katakana words can be expanded to include full forms.

Reviewer: ```suggestion
The `kuromoji_completion` token filter is used to stem Katakana words in Japanese, which are often used to represent foreign words or loanwords. This filter is especially useful for autocompletion or suggest queries, in which partial matches on Katakana words can be expanded to include full forms.
```

> The `kuromoji_completion` token filter is used to stem Katakana words in Japanese, which are often used to represent foreign words or loanwords. This filter is especially useful for autocompletion or suggest queries, in which partial matches on Katakana words can be expanded to include full forms.

Reviewer 2: ```suggestion
The `kuromoji_completion` token filter is used to stem Katakana words in Japanese, which are often used to represent foreign words or loanwords. This filter is especially useful for autocompletion or suggest queries, in which partial matches on Katakana words can be expanded to include their full forms.
```

> 

Reviewer: Line 10, last sentence: "in which partial Katakana word matches can be expanded to their full form"?

Reviewer 2: I think the original is easier to parse.

Reviewer: @kolchfa-aws Ok, but we still need "their" before "full".

> To use this token filter, you must install the `analysis-kuromoji` plugin on all nodes by running `bin/opensearch-plugin install analysis-kuromoji` and restart the cluster. For more information about installing additional plugins, see [Additional plugins]({{site.url}}{{site.baseurl}}/install-and-configure/additional-plugins/index/).

Reviewer: ```suggestion
To use this token filter, you must install the `analysis-kuromoji` plugin on all nodes by running `bin/opensearch-plugin install analysis-kuromoji` and restarting the cluster. For more information about installing additional plugins, see [Additional plugins]({{site.url}}{{site.baseurl}}/install-and-configure/additional-plugins/index/).
```

Reviewer 2: "ing" does not work here because install uses the command and then you restart the cluster. Reworded.

## Example

The following example request creates a new index named `kuromoji_sample` and configures an analyzer with a `kuromoji_completion` filter:

```json
PUT kuromoji_sample
{
  "settings": {
    "index": {
      "analysis": {
        "analyzer": {
          "my_analyzer": {
            "tokenizer": "kuromoji_tokenizer",
            "filter": [
              "my_katakana_stemmer"
            ]
          }
        },
        "filter": {
          "my_katakana_stemmer": {
            "type": "kuromoji_completion"
          }
        }
      }
    }
  }
}
```
{% include copy-curl.html %}

## Generated tokens

> Use the following request to examine the tokens generated using the created analyzer with text which translates to "use a computer":

Reviewer: ```suggestion
Use the following request to examine the tokens generated using the analyzer with text that translates to "use a computer":
```
